sdk/core/azure-core-test/src/main/java/com/azure/core/test/http/PlaybackClient.java

```diff
@@ -111,8 +112,13 @@ public PlaybackClient(RecordedData recordedData, Map<String, String> textReplace
 
             String contentType = networkCallRecord.getResponse().get("Content-Type");
 
-            // octet-stream's are written to disk using Arrays.toString() which creates an output such as "[12, -1]".
-            if (contentType != null && contentType.equalsIgnoreCase("application/octet-stream")) {
+            /*
+             * application/octet-stream and avro/binary are written to disk using Arrays.toString() which creates an
+             * output such as "[12, -1]".
+             */
+            if (contentType != null
+                && (contentType.equalsIgnoreCase(ContentType.APPLICATION_OCTET_STREAM)
+                    || contentType.equalsIgnoreCase("avro/binary"))) {
```

Comment:
It might be better to add "avro/binary" as a constant in [ContentType.java](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core/src/main/java/com/azure/core/http/ContentType.java)